os/fs/vfs/fs_truncate.c

```diff
@@ -180,8 +180,12 @@ int ftruncate(int fd, off_t length)
 	FAR struct file *filep;
 	int ret;
 
-	/* Get the file structure corresponding to the file descriptor. */
+	if (length < 0) {
+		ret = -EINVAL;

```

Comment:
How about assigning the `ret` with positive errno and calling just `set_errno(ret)` in line 204?

Reply:
This is correct, lower level of this will return OK or negative.

Reply:
But this function will return the ERROR, not -ERRORNO. It is just used for setting errno.

Reply:
he understood after I explained, hence resolve it.